Sources/ConfigCat/ConfigService.swift

```diff
@@ -0,0 +1,181 @@
+import Foundation
+
+class ConfigService {
+    private let log: Logger
+    private let fetcher: ConfigFetcher
+    private let mutex: Mutex = Mutex(recursive: true)
+    private let cache: ConfigCache?
+    private let pollingMode: PollingMode
+    private let cacheKey: String
+    private let initialized: Synced<Bool>
```

Comment:
Wrapping this property in `Synced` is perhaps redundant now that a mutex is employed?

Reply:
It was also used outside the boundaries of the mutex, but after a second thought it wasn't a big work to make it protected, so I did that and now it's a regular `Bool`.